csharp/src/Ice/Reference.cs

```diff
@@ -1737,6 +1476,10 @@ private void ClearConnection(Connection connection)
                     // For ice2 the invocation timeout is included in the URI
                     properties[$"{prefix}.InvocationTimeout"] = invocationTimeout.ToPropertyValue();
                 }
+                if (Label is object label)
+                {
+                    properties[$"{prefix}.Label"] = label.ToString() ?? "";
```

Comment:
I would include the Label if its ToString() returns an empty string.
Also TODO: what about escaping this label?

Reply:
do you mean a non-empty string and do the same for the URI option?

Reply:
Sorry, I mean _not_ include this property if ToString() returns an empty string.
As far as escaping, the logical escaping would be the double-quote escaping (`"...."`), not the URI-style percent-escaping.